mithril-aggregator/src/artifact_builder/cardano_database_artifacts/immutable.rs

```diff
@@ -558,46 +533,53 @@ mod tests {
                 .expect_err("Should return an error when an immutable file trio is missing");
         }
 
+        #[test]
+        fn test_retrieve_existing_snapshot_archive() {
+            let work_dir = get_builder_work_dir("return_all_archives_but_not_rebuild_archives");
+            let file_name = "whatever.txt";
+
+            let builder = ImmutableArtifactBuilder::new(
+                work_dir.clone(),
+                vec![Arc::new(MockImmutableFilesUploader::new())],
+                Arc::new(MockSnapshotter::new()),
+                CompressionAlgorithm::Gzip,
+                TestLogger::stdout(),
+            )
+            .unwrap();
+
+            assert_eq!(builder.retrieve_existing_snapshot_archive(file_name), None);
+
+            fs::File::create(work_dir.join(file_name)).unwrap();
+
+            assert_eq!(
+                builder.retrieve_existing_snapshot_archive(file_name),
+                Some(work_dir.join(file_name))
+            );
+        }
+
         #[test]
         fn return_all_archives_but_not_rebuild_archives_already_compressed() {
```

Comment:
It looks like we don't test anymore if the existing archives are recreated in this test.

Reply:
I updated the test so it writes a specific content to the files before running the builder then check that those contents did not change.